runtimevar/driver/driver.go

```diff
@@ -39,20 +39,33 @@ type Variable struct {
 // dictate the type of Variable.Value and a decoding function.  The Watcher provider can use the
 // runtimevar.Decoder to facilitate the decoding logic.
 type Watcher interface {
-	// WatchVariable blocks until the variable changes, the Context's Done channel closes or an
-	// error occurs.
+	// WatchVariable returns one of:
 	//
-	// If the variable has changed, then WatchVariable must return a Variable object with the
-	// new value.
+	// 1. A new value for the variable in v, along with a provider-specific
+	//    version that will be passed to the next WatchVariable call. wait is
+	//    ignored and err must be nil.
+	// 2. A new error. v, version, and wait are ignored.
+	// 3. A nil v and version and err, indicating that the value of the variable
+	//    or the error returned has not changed. WatchVariable will not be called
+	//    again for wait.
```

Comment:
Optional: Seems like this should be a hint more than anything else. There might be reasons (especially in the final form) why the caller might not be able (or want) to respect the time.